src/pages/index.js

```diff
@@ -54,69 +60,229 @@ const cardSection = new Section(
+avatarEditButton.addEventListener("click", () => {
+  avatarPopup.open();
+});
+
+// // Function to handle avatar update
+function handleAvatarSubmit(inputData, saveButton) {
+  renderSaving(true, saveButton);
+  api
+    .updateUserAvatar({ avatar: inputData.avatar })
+    .then((userData) => {
+      userInfor.updateAvatarImage({ avatar: userData.avatar });
+      avatarPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating avatar:", err);
+    })
+    .finally(() => {
+      renderSaving(false, saveButton);
+    });
+}
+
+function renderSaving(isLoading, buttonElement, defaultText = "Save") {
+  if (isLoading) {
+    buttonElement.textContent = "Saving...";
+  } else {
+    buttonElement.textContent = defaultText;
+  }
+}
+
+//////////////////////////////////// end of project 9 ////////////////////////////////////////////
+
+// //----------------------------------------------------------------------------------------
 //                                     Functions
 //----------------------------------------------------------------------------------------
 function createCard(item) {
-  const card = new Card(item, cardSelector, handleImagePreview);
+  const card = new Card(
+    item,
+    cardSelector,
+    handleImagePreview,
+    (cardId, card) => {
+      deleteCardModal(cardId, card);
+    },
+    (cardId, card) => {
+      handleCardLike(cardId, card);
+    }
+  );
   return card.getview();
 }
 
+// function createCard(item) {
+//   const card = new Card(
+//     item,
+//     cardSelector,
+//     handleImagePreview,
+//     handleCardLike,
+//     (cardId, card) => {
+//       deleteCardModal(cardId, card);
+//     },
+//     (cardId, isLiked, cardElement) => {
+//       handleLikeIcon(cardId, isLiked, cardElement, card);
+//     }
+//   );
+//   return card.getview();
+// }
+
+function handleCardLike(card) {
+  if (!card._isLiked) {
+    api
+      .likeCard(card._id)
+      .then((data) => {
+        console.log(data);
+        card.updateLike(true);
+      })
+      .catch((err) => console.error("Error adding like to card:", err));
+  } else {
+    api
+      .dislikeCard(card._id)
+      .then((data) => {
+        card.updateLike(false);
+      })
+      .catch((err) => console.error("Error removing like from card:", err));
+  }
+}
+
 const editFormValidator = new FormValidator(settings, profileEditForm);
 const addFormValidator = new FormValidator(settings, addCardFormElement);
+const avatarFormValidator = new FormValidator(settings, avatarFormElement);
 
 editFormValidator.enableValidation();
 addFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
 
 function handleImagePreview(cardData) {
   imagePreviewPopup.open(cardData);
 }
+// Project 9
+
+function deleteCardModal(cardId, card) {
+  confirmDelete.setSubmitFunction(() => {
+    api
+      .handleDeleteCard(cardId)
+      .then(() => {
+        card._handleDeleteCard();
```

Comment:
`_handleDeleteCard`


* Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.